### PR TITLE
refactor view creator job as a platform service

### PR DESCRIPTION
**Pull request by rleiwang-traceableai**
As a platform service, view creator job can override configuration through Helm chart values.

### CODE REVIEW

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java [hunk 1]
```diff
@@ -1,75 +1,49 @@
...
 
-  private static void updateRuntime() {
-    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-      finalizeLauncher();
-    }));
-  }
-
-  private static void finalizeLauncher() {
+  @Override
+  protected void doStop() {
```
Comment (tim-mwangi) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java:46]: Actually remove the method content for this. It's taken care of over here https://github.com/hypertrace/service-framework/blob/77c248cf0657fe1cf015b0e0b4eae63eedfe9f8c/platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformServiceLauncher.java#L59

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java [hunk 2]
```diff
@@ -1,75 +1,49 @@
...
-        }
-        break;
-      default:
-        System.exit(1);
-        LOGGER.error("Cannot find the class to execute. Did not create the tables");
+  @Override
+  protected void doStart() {
+    try {
+      getTableCreationTool(getAppConfig()).create();
+      shutdown();
```
Comment (tim-mwangi) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java:36]: I'm not sure if you need to explicitly call shutdown(). What happens when you remove this?

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java (hunk 1 above)
Comment (rleiwang-traceableai) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java:46]: done

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java (hunk 2 above)
Comment (rleiwang-traceableai) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java:36]: view creator job stays in running state and installation can't proceed.

Comment (tim-mwangi) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java:29]: oh ok then.

**Pull request merged**